src/v2/view-builder/views/TerminalView.js

```diff
@@ -17,7 +29,7 @@ const Body = BaseForm.extend({
       messagesObjs.value
         .forEach(messagesObj => {
           const msg = messagesObj.message;
-          if (messagesObj.class === 'ERROR') {
+          if (messagesObj.class === 'ERROR' || messagesObj.i18n?.key === 'idx.return.link.expired') {
```

Comment:
@pradeepdewda-okta I think we shall make the error handling more generic like

```
loop all message
  if is error then add error callout
  if is info then add infor callout
  etc..
```

Reply:
@haishengwu-okta I had that in mind, i will create a ticket for this.

Reply:
reuse the `this.options.appState.isExpiredEmailLinkTerminalState`;
you could declare a variable at the beginning of this method (`showMessages`) if you want to avoid duplicated compute.